api/client/service/update.go

```diff
@@ -433,7 +444,11 @@ portLoop:
 }
 
 func equalPort(targetPort nat.Port, port swarm.PortConfig) bool {
-	return (string(port.Protocol) == targetPort.Proto() &&
+	protocol := string(port.Protocol)
+	if protocol == "" {
+		protocol = "tcp"
+	}
+	return (protocol == targetPort.Proto() &&
```

Comment:
Thanks @dnephin! You are right. I was misled by one of the unit tests where the protocol was not specified.
The pull request has been updated to only do a validation of the `--publish-rm`. Please let me know if there are any issues.